get_primary_services/web_app.js

```diff
@@ -0,0 +1,182 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// The test service defined in device_code.js
+const testServiceUUIDs = ['3f2b9742-7dce-11eb-9439-0242ac130002',
+  '6fc096cc-803b-11eb-9439-0242ac130002'];
+// The test characteristics defined in device_code.js
+const firstCharacteristicUUID = '3f2b9c10-7dce-11eb-9439-0242ac130002';
+const secondCharacteristicUUID = 'dcc030e4-8035-11eb-9439-0242ac130002';
+
+/**
+ * Load the device code to the Espruino IDE.
+ */
+function loadEspruinoDeviceCode() {
+  fetch('device_code.js').then(response => response.text()).then(data => {
+    let url = 'http://www.espruino.com/webide?code=' + encodeURIComponent(data);
+    window.open(url, '_window');
+  });
+}
+
+function onGattDisconnected(evt) {
+  const device = evt.target;
+  logInfo(`Disconnected from GATT on device ${device.name}.`);
+}
+
+async function startTest() {
+  clearStatus();
+  logInfo('Starting test');
+
+  $('btn_start_test').disabled = true;
+  $('btn_load_code').disabled = true;
+  let gattServer = undefined;
+  let remoteDevice = undefined;
+
+  const resetTest = async () => {
+    if (gattServer && gattServer.connected) {
+      logInfo('Disconnecting from GATT.');
+      gattServer.disconnect();
+    }
+    $('btn_start_test').disabled = false;
+    $('btn_load_code').disabled = false;
+  }
+
+  // Verify the characteristic properties have the expected values.
+  const verifyProperties = (characteristic, expectedTrue, expectedFalse) => {
+    expectedTrue.forEach(propName => {
+      if (characteristic.properties[propName] !== true)
+        throw `Expected property "${propName}" to be true for characteristic ${characteristic.uuid}`;
+    });
+    expectedFalse.forEach(propName => {
+      if (characteristic.properties[propName] === true)
+        throw `Expected property "${propName}" to be false for characteristic ${characteristic.uuid}`;
+    });
+  };
+
+  // Espruino does not set extended properties, so reliableWrite and
+  // writableAuxiliaries is always false.
+  const alwaysFalseProperties = ['reliableWrite', 'writableAuxiliaries'];
+
+  // Verify the properties of |firstCharacteristicUUID|, which should match
+  // those set in device_code.js.
+  const verifyFirstCharacteristicProperties = (characteristic) => {
+    const trueProps = ['read', 'notify'];
+    const falseProps = ['broadcast', 'writeWithoutResponse', 'write',
+      'indicate', 'authenticatedSignedWrites'].concat(alwaysFalseProperties);
+
+    verifyProperties(characteristic, trueProps, falseProps);
+  };
+
+  // Verify the properties of |secondCharacteristicUUID|, which should match
+  // those set in device_code.js.
+  const verifySecondCharacteristicProperties = (characteristic) => {
+    const trueProps = ['broadcast', 'write', 'writeWithoutResponse', 'indicate'];
+    const falseProps = ['read', 'notify',
+      'authenticatedSignedWrites'].concat(alwaysFalseProperties);
```

Comment:
Add `'authenticatedSignedWrites'` to `alwaysFalseProperties`.

Reply:
done.